pkg/didcomm/envelope/api.go

```diff
@@ -6,12 +6,16 @@ SPDX-License-Identifier: Apache-2.0
 
 package envelope
 
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/packer"
+)
+
 // Package envelope manages the handling of DIDComm raw messages in JWE compliant envelopes.
 // The aim of this package is build these envelopes and parse them. They are mainly used as
 // wire-level wrappers of 'payloads' used in DID Exchange flows.
 
 // PackagerCreator method to create new outbound dispatcher service
-type PackagerCreator func(prov Provider) (Packager, error)
+type PackagerCreator func(prov PackerProvider) (Packager, error)
 
 // Packager provide methods to pack and unpack msg
 type Packager interface {
```

Comment:
now that Crypter is renamed to Packer, this Packager interface becomes an extra layer that we can skim.

Reply:
It now has the purpose of selecting which pack/unpack algorithm to use, allowing the agent to support communicating using multiple message formats for compatibility - so it's not a layer we can pull out.